### ISSUE TITLE
walk.lua:30: Invalid value for parameter 'x'

### ISSUE THREAD

**Issue by AlexUrbanAc** [CorsixTH/CorsixTH#1394]
**CorsixTH Version:** 0.61

**Operating System:** Windows 10
**Theme Hospital Version:** GOG.com

#### Describe the issue
Evacuation of the hospital due to uncontained epidemic caused an assert error.

#### Gamelog.txt
Savegame version is 122 (v0.61), originally it was 122 (v0.61)
Warning: goHome called when the patient is already going home
Warning: goHome called when the patient is already going home
Warning: Trying to remove nonexistant room build callback (function: 38EA7C00) from humanoid (table: 3C9AFC58).
Warning: Trying to remove nonexistant room remove callback (function: 3DE1A2F0) from humanoid (table: 3C9AFC58).
Warning: goHome called when the patient is already going home
Warning: goHome called when the patient is already going home
Error in timer handler: 
D:\CorsixTH\Lua\humanoid_actions\walk.lua:30: Invalid value for parameter 'x'
stack traceback:
	[C]: in function 'assert'
	D:\CorsixTH\Lua\humanoid_actions\walk.lua:30: in local 'constructor'
	D:\CorsixTH\Lua\class.lua:74: in global 'WalkAction'
	D:\CorsixTH\Lua\entities\humanoid.lua:727: in method 'walkTo'
	D:\CorsixTH\Lua\rooms\psych.lua:108: in method 'loop_callback'
	D:\CorsixTH\Lua\humanoid_actions\use_object.lua:324: in local 'timer_function'
	D:\CorsixTH\Lua\entity.lua:216: in field 'tick'
	D:\CorsixTH\Lua\entities\staff.lua:149: in method 'tick'
	D:\CorsixTH\Lua\world.lua:1051: in method 'onTick'
	D:\CorsixTH\Lua\app.lua:1008: in function <D:\CorsixTH\Lua\app.lua:1005>
	(...tail calls...)
	D:\CorsixTH\Lua\app.lua:909: in function <D:\CorsixTH\Lua\app.lua:904>

**Comment by sadger** [comment]
Do you have a save before this occurred by any chance?

**Comment by AlexUrbanAc** [comment]
Yes, must have forgotten to upload it.
[Ulti XT.zip](https://github.com/CorsixTH/CorsixTH/files/2034833/Ulti.XT.zip)

**Comment by TheCycoONE** [comment]
Did you do anything with the screen in the psych room?

This error would happen if there was no screen in the room, if it was inaccessible, or possibly if it was in a corrupt state.

**Comment by AlexUrbanAc** [comment]
I think all eight psych rooms worked before the evacuation, and I didn't touch anything during the evacuation. I wanted to wait until the hospital is empty before doing anything, but the game crashes before the evacuation is completed.

**Comment by mugmuggy** [comment]
I wrote a function to call the same code as Epidemic:evacuate. In doing so I was able to replicate the issues without having an active Epidemic.

The loop_callback in PsychRoom:commandEnteringPatient is executing endlessly, as the patient had kings_complex, the walkTo and UseScreen actions are added to the end of the action queue - repeatedly, so as the patient leaves the room, the walkTo is again being called but this time no object could be found.

Changing just https://github.com/CorsixTH/CorsixTH/blob/b8e9d7ff500f357b8e7a6562e4aaec22c48ee102/CorsixTH/Lua/rooms/psych.lua#L97 to an equality check will prevent the repeatedness and fixes the issue.

However will leave the doctor in processing the callback until another patient arrives.
